internal/datanode/writebuffer/write_buffer.go

```diff
@@ -162,6 +166,40 @@ func (wb *writeBufferBase) GetFlushTimestamp() uint64 {
 	return wb.flushTimestamp.Load()
 }
 
+func (wb *writeBufferBase) MemorySize() int64 {
+	wb.mut.RLock()
+	defer wb.mut.RUnlock()
+
+	var size int64
+	for _, segBuf := range wb.buffers {
+		size += segBuf.MemorySize()
+	}
+	return size
+}
+
+func (wb *writeBufferBase) EvictBuffer(policies ...SyncPolicy) {
+	wb.mut.Lock()
+	defer wb.mut.Unlock()
+
+	log := log.Ctx(context.Background()).With(
```

Comment:
why using background log? should the syncSegments ctx be the same with this ctx?

Reply:
EvictBuffer & SealSegment are different API. EvictBuffer does not have context here